compiler/rustc_symbol_mangling/src/export.rs

```diff
@@ -0,0 +1,191 @@
+use std::assert_matches::debug_assert_matches;
+
+use rustc_abi::IntegerType;
+use rustc_data_structures::stable_hasher::StableHasher;
+use rustc_hashes::Hash128;
+use rustc_hir::def::DefKind;
+use rustc_middle::ty::{self, Instance, Ty, TyCtxt};
+use rustc_span::symbol::{Symbol, sym};
+
+trait AbiHashStable<'tcx> {
+    fn hash(&self, tcx: TyCtxt<'tcx>, hasher: &mut StableHasher);
```

Comment:
Does this need to be a trait? It's not currently used generically.

Also, probably better to rename the method to something like `abi_hash` to avoid confusing it with regular `hash`.

Reply:
I keep trait implementations in a single file, since the feature is unstable. Ideally, `#[derive]` should be used.